pkg/monitor/controller.go

```diff
@@ -173,6 +237,38 @@ func (monitor *controllerMonitor) deleteStaleAgentCRD(old interface{}) {
 	monitor.deleteAgentCRD(node.Name)
 }
 
+func (monitor *controllerMonitor) createExternalNodeAgentCRD(obj interface{}) {
+	en := obj.(*v1alpha1.ExternalNode)
+	klog.Infof("Creating ExternalNode agent monitoring CRD %s", en.Name)
+	agentCRD := new(v1beta1.AntreaAgentInfo)
+	agentCRD.Name = en.Name
+	_, err := monitor.client.CrdV1beta1().AntreaAgentInfos().Create(context.TODO(), agentCRD, metav1.CreateOptions{})
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			klog.Infof("Skipping creating agent monitoring CRD %s as it already exists", en.Name)
+		} else {
+			klog.Errorf("Failed to create agent monitoring CRD %s: %v", en.Name, err)
+		}
+	}
+}
+
+func (monitor *controllerMonitor) deleteExternalNodeAgentCRD(old interface{}) {
+	en, ok := old.(*v1alpha1.ExternalNode)
+	if !ok {
+		tombstone, ok := old.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.Errorf("Error decoding object when deleting ExternalNode, invalid type: %v", old)
+			return
+		}
+		en, ok = tombstone.Obj.(*v1alpha1.ExternalNode)
+		if !ok {
+			klog.Errorf("Error decoding object tombstone when deleting ExternalNode, invalid type: %v", tombstone.Obj)
+			return
+		}
+	}
+	monitor.deleteAgentCRD(en.Name)
+}
+
 func (monitor *controllerMonitor) deleteAgentCRD(name string) {
 	klog.Infof("Deleting agent monitoring CRD %s", name)
 	err := monitor.client.CrdV1beta1().AntreaAgentInfos().Delete(context.TODO(), name, metav1.DeleteOptions{})
```

Comment:
Maybe you can check the not found error and ignore it too (not to log an error).

Reply:
Got it, have updated this part accordingly.